### PR TITLE
DPI scaling fix in Direct2D engine, ClipBounds property fix in Direct2D and WinForms

### PR DESCRIPTION
**Pull request by e673**
It's a fix to an issue #1141.

1. Direct2D: `ClipBounds` value was incorrect when DPI scaling was on due to rounding and logical problems.
2. WinForms: `ClipBounds` should not depend on PixelOffsetMode value.
3. Direct2D: since the control size is physical size regardess of screen DPI, there is no need to scale WicRenderTarget.

### CODE REVIEW

File: src/Eto.WinForms/Drawing/GraphicsHandler.cs [hunk 1]
```diff
@@ -425,7 +425,11 @@ public IMatrix CurrentTransform
 
 		public RectangleF ClipBounds
 		{
-			get { return this.Control.ClipBounds.ToEto(); }
+			get
+			{
+				this.Control.PixelOffsetMode = sdd.PixelOffsetMode.None;
+				return this.Control.ClipBounds.ToEto();
```
Comment (cwensley) [inline src/Eto.WinForms/Drawing/GraphicsHandler.cs]: I think you need to restore the previous mode after getting the bounds here.

**Comment by cwensley** [comment]
Hey @e673, this is great thanks for the PR!  Looking good except for restoring the PixelOffsetMode on WinForms when getting the ClipBounds.

**Comment by e673** [comment]
> Hey @e673, this is great thanks for the PR! Looking good except for restoring the PixelOffsetMode on WinForms when getting the ClipBounds.

Yes, I was thinking about it. But this value is updated in every call of any drawing method (DrawLine, FillRectangle etc) by calling `SetOffset(...)`. So I see no sense in restoring its value.

**Comment by cwensley** [comment]
Awesome, thanks for the contribution!!

**Pull request merged**